### PR TITLE
feat: Add article about closure objects

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

File: blog/2022-10-17-closure-objects.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+---
+slug: closure-objects
+title: How to remove Java from JavaScript.
```
Comment (marketachalupnikova) [inline blog/2022-10-17-closure-objects.md:3]: delete comma

Comment (stepan662) [reply blog/2022-10-17-closure-objects.md]: 👍

File: blog/2022-10-17-closure-objects.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+authors: [sgranat]
+image: '/img/blog/closure-objects/js-without-java.png'
+description: Replacing JavaScript classes with closure and simple JS constructions.
```
[the remark this answers is not in the document]
Comment (stepan662) [reply blog/2022-10-17-closure-objects.md]: 👍

File: blog/2022-10-17-closure-objects.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+title: How to remove Java from JavaScript
+authors: [sgranat]
+image: '/img/blog/closure-objects/js-without-java.png'
+description: Replacing JavaScript classes with closure and simple JS constructions. Write React-like code even without React.
+tags: [JavaScript, class, closure, React, TypeScript]
+---
+
+As a React developer, I have an aversion to JS classes. In React bubble, it feels like a bad practice when I see them somewhere.
+
+However, recently I got a task to rewrite our `@tolgee/core`, which is VanillaJS, and so far, there have been classes everywhere. I couldn't stand it, we shouldn't use classes, that's bad. But then I realized that I don't actually know how to replace classes outside of React. Hmm ...
```
Comment (JanCizmar) [inline blog/2022-10-17-closure-objects.md:12]: So you are basically saying. I made my opinion on wrong arguments. I hated classes, but then I found out I need them, so I started to look for a way how to replace them with something not-classy, to be able to stay with my opinion. This is not a good way how to advocate your solution.

You should precisely describe, why you consider classes bad practice at first. Not just describe how to replace them.

**Review by JanCizmar — CHANGES_REQUESTED** [review]

File: blog/2022-10-17-closure-objects.md (hunk 3 above)
Comment (JanCizmar) [reply blog/2022-10-17-closure-objects.md:12]: You can keep it agressive. But there is actually no explanation why classes are bad in the article except the Crockfords citation and that you don't want to write keywords like Java guys, which is both bullshit. And Crokford is criticised a lot for that. You can use it as a joke, but in the same time you should advocate why you consider these concepts wrong. Why I should stop using Classes when I don't hate Java?

File: blog/2022-10-17-closure-objects.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    setSpeed,
+  });
+}
+```
+
+I've always found class inheritance confusing, and this way is much more readable for me. It's also quite similar to how React allows you to compose hooks, so it feels familiar.
+
+Even though I haven't seen this method used in many places, it feels quite natural to me. I like how I see exactly what is passed where. It also brings an advantage in code optimization as private methods are truly private (unlike `Typescript` private methods), so the optimizer can optimize them accordingly.
+
+One simple advantage is that you can safely use the spread operator here, as we won't run into issues with binding and prototype chain.
```
[the remark this answers is not in the document]
Comment (stepan662) [reply blog/2022-10-17-closure-objects.md:104]: 👍

File: blog/2022-10-17-closure-objects.md (hunk 3 above)
Comment (stepan662) [reply blog/2022-10-17-closure-objects.md]: Ok

Comment (stepan662) [reply blog/2022-10-17-closure-objects.md]: I've added an section about this

File: blog/2022-10-17-closure-objects.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+<!--truncate-->
+
+## Wait, what's wrong with the classes?
+
+Classes are basically added syntax sugar which gives you the impression that you are working with a class based programming language when you are actually not. Classes in JavaScript are implemented differently from the others. My main complaints are these:
+
+- How `this` works is a cause of many issues, and correctly understanding it is not easy
+- Inheritance is becoming an outdated concept, and classes encourage you to use it
+- Private fields in classes are not supported everywhere yet, and their syntax is weird
```
Comment (JanCizmar) [inline blog/2022-10-17-closure-objects.md:22]: and typescript private fields are not really private

**Review by JanCizmar — APPROVED** [review]

File: blog/2022-10-17-closure-objects.md (hunk 5 above)
Comment (stepan662) [reply blog/2022-10-17-closure-objects.md]: 👍

**Pull request merged**